File: Binary Tree/226.翻转二叉树.cpp
#include <queue>

using namespace std;
/*
 * @lc app=leetcode.cn id=226 lang=cpp
 *
 * [226] 翻转二叉树
 */

// @lc code=start
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution
{
public:
    TreeNode *invertTree(TreeNode *root)
    {
        queue<TreeNode *> qu;
        if (root != nullptr)
        {
            qu.push(root);
        }

        while (!qu.empty())
        {
            TreeNode *cur = qu.front();
            qu.pop();

            if (cur != nullptr)
            {
                TreeNode *temp = cur->left;
                cur->left = cur->right;
                cur->right = temp;

                qu.push(cur->left);
                qu.push(cur->right);
            }
        }

        return root;
    }
};
// @lc code=end

File: Binary Tree/二叉树的修改与构造/226.翻转二叉树.ts
/*
 * @lc app=leetcode.cn id=226 lang=typescript
 *
 * [226] 翻转二叉树
 */

// @lc code=start
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */
// 直接递归实现
function invertTree(root: TreeNode | null): TreeNode | null {
  const exchangeTree = (tree: TreeNode | null): void => {
    if (!tree) {
      return
    }

    let temp = tree.left
    tree.left = tree.right
    tree.right = temp

    exchangeTree(tree.left)
    exchangeTree(tree.right)
  }

  exchangeTree(root)

  return root
};
// @lc code=end

